modules/incentives/src/tests.rs

```diff
@@ -418,68 +418,183 @@ fn payout_works() {
+		assert_eq!(TokensModule::free_balance(DOT, &VAULT::get()), 0);
+		assert_eq!(TokensModule::free_balance(AUSD, &BOB::get()), 18);
+		assert_eq!(TokensModule::free_balance(DOT, &BOB::get()), 20);
 		assert_eq!(
 			RewardsModule::pool_infos(PoolId::Loans(BTC)),
 			PoolInfo {
-				total_shares: 200,
-				rewards: vec![(AUSD, (27, 22))].into_iter().collect(),
+				total_shares: 300,
+				rewards: vec![(AUSD, (27, 27)), (DOT, (30, 30))].into_iter().collect(),
 			}
 		);
 		assert_eq!(
 			RewardsModule::shares_and_withdrawn_rewards(PoolId::Loans(BTC), BOB::get()),
-			(100, vec![(AUSD, 13)].into_iter().collect())
+			(200, vec![(AUSD, 18), (DOT, 20)].into_iter().collect())
+		);
+		assert_eq!(
+			IncentivesModule::pending_multi_rewards(PoolId::Loans(BTC), BOB::get()),
+			Default::default()
+		);
+
+		// accumulate 6 aUSD
+		assert_ok!(TokensModule::deposit(AUSD, &VAULT::get(), 6));
+		assert_ok!(RewardsModule::accumulate_reward(&PoolId::Loans(BTC), AUSD, 6));
+		assert_eq!(TokensModule::free_balance(AUSD, &VAULT::get()), 15);
+		assert_eq!(
+			RewardsModule::pool_infos(PoolId::Loans(BTC)),
+			PoolInfo {
+				total_shares: 300,
+				rewards: vec![(AUSD, (33, 27)), (DOT, (30, 30))].into_iter().collect(),
+			}
+		);
+
+		// Alice claim AUSD reward
+		assert_ok!(IncentivesModule::claim_rewards(
+			Origin::signed(ALICE::get()),
+			PoolId::Loans(BTC)
+		));
+		System::assert_last_event(Event::IncentivesModule(crate::Event::ClaimRewards {
+			who: ALICE::get(),
+			pool: PoolId::Loans(BTC),
+			reward_currency_id: AUSD,
+			actual_amount: 11,
+			deduction_amount: 0,
+		}));
+
+		assert_eq!(TokensModule::free_balance(AUSD, &VAULT::get()), 4);
+		assert_eq!(TokensModule::free_balance(AUSD, &ALICE::get()), 11);
+		assert_eq!(
+			RewardsModule::pool_infos(PoolId::Loans(BTC)),
+			PoolInfo {
+				total_shares: 300,
+				rewards: vec![(AUSD, (33, 29)), (DOT, (30, 30))].into_iter().collect(),
+			}
+		);
+		assert_eq!(
+			RewardsModule::shares_and_withdrawn_rewards(PoolId::Loans(BTC), ALICE::get()),
+			(100, vec![(AUSD, 11), (DOT, 10)].into_iter().collect())
+		);
+		assert_eq!(
+			IncentivesModule::pending_multi_rewards(PoolId::Loans(BTC), ALICE::get()),
+			Default::default()
+		);
+
+		// mock the Vault is enough for some reasons
+		assert_ok!(TokensModule::withdraw(AUSD, &VAULT::get(), 3));
+		assert_eq!(TokensModule::free_balance(AUSD, &VAULT::get()), 1);
+
+		// Bob claim rewards, payout AUSD failed for drained vault, the pending reward record of AUSD will
+		// not change.
+		assert_ok!(IncentivesModule::claim_rewards(
+			Origin::signed(BOB::get()),
+			PoolId::Loans(BTC)
+		));
+
+		assert_eq!(TokensModule::free_balance(AUSD, &VAULT::get()), 1);
+		assert_eq!(TokensModule::free_balance(AUSD, &BOB::get()), 18);
+		assert_eq!(
+			RewardsModule::pool_infos(PoolId::Loans(BTC)),
+			PoolInfo {
+				total_shares: 300,
+				rewards: vec![(AUSD, (33, 33)), (DOT, (30, 30))].into_iter().collect(),
```

Comment:
above claim failed so I think reward for AUSD should be `(33, 27)` cause 4 is still `pending_multi_rewards`

Reply:
`claim_rewards` will trigger `rewards.claim_rewards` first. 
It will settle new accumulated rewards(4 aUSD) for BOB,  then added it to the `incentives.PendingMultiRewards` of BOB (The withdrawn record of `rewards.PoolInfos` and BOB's `rewards.SharesAndWithdrawn` will also added the new settled rewards)

After settled,  will trying to payout the `PendingMultiRewards` to BOB.  But this step failed due to aUSD transfer failure,  BOB's `PendingMultiRewards` will not change at this step,  still keep 4 aUSD.